src/main/java/com/google/sps/servlets/ListDrugInfoServlet.java

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.datastore.*;
+import com.google.gson.Gson;
+import com.google.sps.data.DrugDetails;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+//Servlet responsible for getting drug info.
+
+@WebServlet("/list-drug-info")
+public class ListDrugInfoServlet extends HttpServlet {
+
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    String Name = request.getParameter("Name");
+    Query<Entity> query = Query.newEntityQueryBuilder()
+            .setKind("Drug")
+            .setFilter(StructuredQuery.PropertyFilter.eq("Name", Name))
```

Comment:
I'm still seeing a query by name, not a lookup by key ID which is simpler and more efficient.